mParticle-ROKOMobi/MPKitROKOMobi.m

```diff
@@ -0,0 +1,207 @@
+    [self.proxy.getLinkManager continueUserActivity:userActivity];
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceROKOMobi) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (nonnull MPKitExecStatus *)openURL:(nonnull NSURL *)url options:(nullable NSDictionary<NSString *, id> *)options {
+    [self.proxy.getLinkManager handleDeepLink:url];
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceROKOMobi) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (nonnull MPKitExecStatus *)openURL:(nonnull NSURL *)url sourceApplication:(nullable NSString *)sourceApplication annotation:(nullable id)annotation {
+    [self.proxy.getLinkManager handleDeepLink:url];
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceROKOMobi) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+#pragma mark Push
+
+- (MPKitExecStatus *)handleActionWithIdentifier:(NSString *)identifier forRemoteNotification:(NSDictionary *)userInfo {
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceROKOMobi) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (MPKitExecStatus *)receivedUserNotification:(NSDictionary *)userInfo {
+    if (_pusher) {
+        [_pusher handleNotification:userInfo];
+    }
+    
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceROKOMobi) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (MPKitExecStatus *)setDeviceToken:(NSData *)deviceToken {
+    _pusher = [[ROKOPush alloc]init];
+    [_pusher registerWithAPNToken:deviceToken withCompletion:^(id responseObject, NSError *error) {
+        if (error) NSLog(@"Failed to register with error - %@", error);
+    }];
+    
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceROKOMobi) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+#pragma mark User attributes and identities
+
+- (MPKitExecStatus *)setUserAttribute:(NSString *)key value:(NSString *)value {
+    ROKOPortalManager *portalManager = [ROKOComponentManager sharedManager].portalManager;
+    
+    [portalManager setUserCustomProperty:value forKey:key completionBlock:^(NSError * _Nullable error) {
+        if (error) NSLog(@"%@", error);
+    }];
+    
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceROKOMobi) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (MPKitExecStatus *)setUserIdentity:(NSString *)identityString identityType:(MPUserIdentity)identityType {
+    ROKOPortalManager *portalManager = [ROKOComponentManager sharedManager].portalManager;
+     
+    [portalManager setUserWithName:identityString referralCode:nil linkShareChannel:nil completionBlock:^(NSError * _Nullable error) {
```

Comment:
You should pick a `MPUserIdentity` (likely either email or customerid) and only call your API if this is the identity passed. Otherwise, apps that set both will have a non-deterministic mapping to your API.